src/darwin/Framework/CHIP/MTRDeviceControllerStartupParams.mm

```diff
@@ -106,6 +135,44 @@ - (instancetype)initWithParams:(MTRDeviceControllerStartupParams *)params
     _operationalKeypair = params.operationalKeypair;
     _operationalCertificateIssuer = params.operationalCertificateIssuer;
     _operationalCertificateIssuerQueue = params.operationalCertificateIssuerQueue;
+    _uniqueIdentifier = params.uniqueIdentifier;
+
+    return self;
+}
+
+- (instancetype)initWithParameters:(MTRDeviceControllerStartupParameters *)params error:(CHIP_ERROR &)error
+{
+    if (!(self = [super init])) {
+        error = CHIP_ERROR_INCORRECT_STATE;
+        return nil;
+    }
+
+    if (![params isKindOfClass:[MTRDeviceControllerExternalCertificateStartupParameters class]]) {
+        MTR_LOG_ERROR("Unexpected subclass of MTRDeviceControllerStartupParameters");
+        error = CHIP_ERROR_INVALID_ARGUMENT;
+        return nil;
+    }
+
+    _nocSigner = nil;
+
+    NSNumber * fabricID;
+    error = ExtractFabricIDFromNOC(params.operationalCertificate, &fabricID);
```

Comment:
any reason why nodeid is not extracted from the NOC here?

Reply:
Ah, this is pretty subtle and should be documented.  I will do that.